post_processing/utils/metadata_extractor.py

```diff
@@ -0,0 +1,569 @@
+                    key, value = line.split('=', 1)
+                    key = key.strip()
+                    value = value.strip()
+
+                    if key in key_params:
+                        params[key] = value
+
+        except Exception as e:
+            logger.warning(f"Failed to parse sysctl: {str(e)}")
+
+        return params
+
+    def _parse_kernel_cmdline(self, cmdline: str) -> Dict[str, Any]:
+        """
+        Parse kernel command line into structured parameters
+
+        Example:
+            "console=tty0 console=hvc0 rw ostree=/path"
+            -> {
+                "console": ["tty0", "hvc0"],
+                "rw": true,
+                "ostree": "/path"
+            }
+        """
+        params = {}
+
+        # Split by spaces
+        parts = cmdline.split()
+
+        for part in parts:
+            if '=' in part:
+                key, value = part.split('=', 1)
+                # Handle multiple values for same key
+                if key in params:
+                    # Convert to list if not already
+                    if not isinstance(params[key], list):
+                        params[key] = [params[key]]
+                    params[key].append(value)
+                else:
+                    params[key] = value
+            else:
+                # Boolean flag (no value)
+                params[part] = True
+
+        # Keep only the most useful params (not the giant paths)
+        useful_keys = [
+            'console', 'rw', 'ro', 'quiet', 'splash', 'rhgb',
+            'selinux', 'enforcing', 'nofb', 'nomodeset',
+            'intel_iommu', 'amd_iommu', 'iommu',
+            'hugepages', 'hugepagesz', 'default_hugepagesz',
+            'numa', 'transparent_hugepage', 'elevator',
+            'isolcpus', 'nohz', 'nohz_full', 'rcu_nocbs'
+        ]
+
+        filtered_params = {}
+        for key in useful_keys:
+            if key in params:
+                filtered_params[key] = params[key]
+
+        # Add count of total parameters
+        filtered_params['_total_parameters'] = len(params)
+
+        return filtered_params
+
+    @staticmethod
+    def _parse_memory_size(size_str: str) -> int:
+        """
+        Parse memory size string to GB
+
+        Examples:
+            "128G" -> 128
+            "256000M" -> 256
+            "1T" -> 1024
+        """
+        if not size_str:
+            return 0
+
+        size_str = size_str.strip().upper()
+
+        # Extract number and unit
+        match = re.match(r'([\d.]+)\s*([KMGT])?', size_str)
+        if not match:
+            return 0
+
+        value = float(match.group(1))
+        unit = match.group(2) if match.group(2) else 'M'
+
+        # Convert to GB
+        if unit == 'K':
+            return round(value / 1024 / 1024)
+        elif unit == 'M':
+            return round(value / 1024)
+        elif unit == 'G':
+            return round(value)
+        elif unit == 'T':
+            return round(value * 1024)
```

Comment:
Are we getting IEC base-2 amounts or SI base-10 amounts?

Mix/matching them would be probably close enough:tm:, but could provide headaches.